packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,326 @@
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-containerOverrides
+   *
+   * @default - No container overrides
+   */
+  readonly containerOverrides?: ContainerOverrides;
+
+  /**
+   * A list of dependencies for the job.
+   * A job can depend upon a maximum of 20 jobs.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-dependsOn
+   *
+   * @default - No dependencies
+   */
+  readonly dependsOn?: JobDependency[];
+
+  /**
+   * The payload to be passed as parametrs to the batch job
+   *
+   * @default - No parameters are passed
+   */
+  readonly payload?: { [key: string]: any };
+
+  /**
+   * The number of times to move a job to the RUNNABLE status.
+   * You may specify between 1 and 10 attempts.
+   * If the value of attempts is greater than one,
+   * the job is retried on failure the same number of attempts as the value.
+   *
+   * @default - 1
+   */
+  readonly attempts?: number;
+
+  /**
+   * The timeout configuration for this SubmitJob operation.
+   * The minimum value for the timeout is 60 seconds.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/APIReference/API_SubmitJob.html#Batch-SubmitJob-request-timeout
+   *
+   * @default - No timeout
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The service integration pattern indicates different ways to call TerminateCluster.
+   *
+   * The valid value is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default SYNC
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+}
+
+/**
+ * A Step Functions Task to run AWS Batch
+ */
+export class RunBatchJob implements sfn.IStepFunctionsTask {
+  private readonly integrationPattern: sfn.ServiceIntegrationPattern;
+
+  constructor(private readonly props: RunBatchJobProps) {
+    // validate integrationPattern
+    this.integrationPattern =
+      props.integrationPattern || sfn.ServiceIntegrationPattern.SYNC;
+
+    const supportedPatterns = [
+      sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+      sfn.ServiceIntegrationPattern.SYNC
+    ];
+
+    if (!supportedPatterns.includes(this.integrationPattern)) {
+      throw new Error(
+        `Invalid Service Integration Pattern: ${this.integrationPattern} is not supported to call RunBatchJob.`
+      );
+    }
+
+    // validate arraySize limits
+    if (
+      props.arraySize !== undefined &&
+      (props.arraySize < 2 || props.arraySize > 10000)
+    ) {
+      throw new Error(
+        `Invalid value of arraySize. The array size can be between 2 and 10,000.`
+      );
+    }
+
+    // validate dependency size
+    if (props.dependsOn && props.dependsOn.length > 20) {
+      throw new Error(
+        `Invalid number of dependencies. A job can depend upon a maximum of 20 jobs.`
+      );
+    }
+
+    // validate attempts
+    if (
+      props.attempts !== undefined &&
+      (props.attempts < 1 || props.attempts > 10)
+    ) {
```

Comment:
Same here. Keeping all of this on a single line is better.

I'm not fussed on this however, if you prefer to leave it this way.

Reply:
Same as above, Prettier is formatting it to multiple lines.